src/pages/Search/SearchFiltersDatePage.tsx

```diff
@@ -1,16 +1,38 @@
+import {format} from 'date-fns';
 import React from 'react';
 import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
 import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
+import DatePicker from '@components/DatePicker';
+import FormProvider from '@components/Form/FormProvider';
+import InputWrapper from '@components/Form/InputWrapper';
+import type {FormOnyxValues} from '@components/Form/types';
 import HeaderWithBackButton from '@components/HeaderWithBackButton';
 import ScreenWrapper from '@components/ScreenWrapper';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
-import Text from '@src/components/Text';
+import {mergeFilters} from '@libs/actions/Search';
+import Navigation from '@libs/Navigation/Navigation';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import INPUT_IDS from '@src/types/form/SearchAdvancedFiltersForm';
+
+const DEFAULT_DATE = format(new Date(), CONST.DATE.FNS_FORMAT_STRING);
+const DEFAULT_FILTERS_FORM_VALUE = {[INPUT_IDS.DATE_AFTER]: DEFAULT_DATE, [INPUT_IDS.DATE_BEFORE]: DEFAULT_DATE};
 
 function SearchFiltersDatePage() {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
 
+    const [searchAdvancedFiltersForm = DEFAULT_FILTERS_FORM_VALUE] = useOnyx(ONYXKEYS.FORMS.SEARCH_ADVANCED_FILTERS_FORM);
+    const defaultDateAfter = searchAdvancedFiltersForm[INPUT_IDS.DATE_AFTER];
+    const defaultDateBefore = searchAdvancedFiltersForm[INPUT_IDS.DATE_BEFORE];
```

Comment:
I don't think we should add defaults here. All of these values are optional so the default should be no date selected

Reply:
If we want default date to not be set, default values should be changed from current date to undefined, but I need to create them so that Typescript is assured that `searchAdvancedFiltersForm` is not undefined.

Reply:
If I apply these changes, TS still isn't sure the whole object under Onyx key isn't undefined:
<img width="896" alt="351312560-6c214aad-7a1f-45c3-af2b-7806442c04df" src="https://github.com/user-attachments/assets/a2e96986-b3f1-4a06-83ec-5af3fc604065">
Alternatively, I can assign empty object with a cast, that works as well:
`const [searchAdvancedFiltersForm = {} as SearchAdvancedFiltersForm] = useOnyx(ONYXKEYS.FORMS.SEARCH_ADVANCED_FILTERS_FORM)`
`defaultDateAfter/Before` are both extracted just for clarity

Reply:
Good point, optional chaining went through my mind, but I thought I would be less clean, I'll change it right now